pkg/repositories/models/dataset.go

```diff
@@ -20,3 +25,12 @@ type PartitionKey struct {
 	DatasetUUID string `gorm:"type:uuid;primary_key"`
 	Name        string `gorm:"primary_key"`
 }
+
+// BeforeCreate so that we set the UUID in golang rather than from a DB function call
+func (datasetKey *DatasetKey) BeforeCreate(tx *gorm.DB) error {
+	if datasetKey.UUID == "" {
```

Comment:
isn't this an error if its already assigned? (not sure if too defensive of programming to actually check)

Reply:
maybe.